src/libusermode/userhook.cpp

```diff
@@ -415,8 +418,14 @@ static event_response_t perform_hooking(drakvuf_t drakvuf, drakvuf_trap_info * i
     event_response_t ret = internal_perform_hooking(drakvuf, info, plugin, dll_meta);
 
     if (!was_hooked && dll_meta->v.is_hooked) {
+        std::vector<hook_target_view_t> targets;
+
+        for (auto& target : dll_meta->targets) {
+            targets.emplace_back(target.target_name, target.offset, target.state);
+        }
+
         for (auto& reg : plugin->plugins) {
-            reg.post_cb(drakvuf, (const dll_view_t*)dll_meta, reg.extra);
+            reg.post_cb(drakvuf, (const dll_view_t*)dll_meta, (const std::vector<hook_target_view_t> *)&targets, reg.extra);
```

Comment:
This cast makes me really worried. If code doesn't build without it then it's usually a pretty good indication that something went really wrong here.